src/clib/pio_rearrange.c

```diff
@@ -984,6 +968,7 @@ void determine_fill(iosystem_desc_t ios, io_desc_t *iodesc, const int gsize[],
 
     MPI_Allreduce(MPI_IN_PLACE, &totalllen, 1, PIO_OFFSET, MPI_SUM, ios.union_comm);
 
+    /* ??? */
```

Comment:
Did you see my other comment about how you should not be doing this with netCDF-4 files?